variables.tf

```
@@ -33,70 +43,57 @@ variable "is_managed" {
   default     = false
 }
 
-variable "create_default_roles" {
-  description = "Whether the default roles should be created"
-  type        = bool
-  default     = false
-}
-
-variable "add_grants_to_existing_objects" {
-  description = "Whether grants to existing objects should be added"
+variable "create_default_database_roles" {
+  description = "Whether the default database roles should be created"
   type        = bool
   default     = false
 }
 
 variable "roles" {
   description = "Roles created in the scheme scope"
   type = map(object({
-    enabled                        = optional(bool, true)
-    comment                        = optional(string)
-    role_ownership_grant           = optional(string)
-    granted_roles                  = optional(list(string))
-    granted_to_roles               = optional(list(string))
-    granted_to_users               = optional(list(string))
-    add_grants_to_existing_objects = optional(bool)
-    schema_grants                  = optional(list(string))
-    table_grants                   = optional(list(string))
-    dynamic_table_grants           = optional(list(string))
-    external_table_grants          = optional(list(string))
-    view_grants                    = optional(list(string))
-    materialized_view_grants       = optional(list(string))
-    file_format_grants             = optional(list(string))
-    function_grants                = optional(list(string))
-    stage_grants                   = optional(list(string))
-    task_grants                    = optional(list(string))
-    procedure_grants               = optional(list(string))
-    sequence_grants                = optional(list(string))
-    stream_grants                  = optional(list(string))
+    enabled                   = optional(bool, true)
+    role_ownership_grant      = optional(string)
+    granted_to_roles          = optional(list(string))
+    granted_to_database_roles = optional(list(string))
+    granted_database_roles    = optional(list(string))
+    database_grants           = optional(list(string))
```

based on:
https://github.com/getindata/terraform-snowflake-database-role/blob/96bf8f6a78443489a39f7b8883dba3c0ff9f06d2/variables.tf#L41-L48
```suggestion
    database_grants           = optional(object({
      all_privileges    = optional(bool)
      with_grant_option = optional(bool, false)
      privileges        = optional(list(string), null)
    }))
```